presto-main/src/main/java/com/facebook/presto/operator/repartition/ArrayBlockEncodingBuffer.java

```diff
@@ -239,8 +239,8 @@ protected void setupDecodedBlockAndMapPositions(DecodedBlockNode decodedBlockNod
         double targetBufferSize = partitionBufferCapacity * decodedBlockPageSizeFraction *
                 (estimatedSerializedSizeInBytes - childrenEstimatedSerializedSizeInBytes) / estimatedSerializedSizeInBytes;
 
-        setEstimatedNullsBufferMaxCapacity((int) (targetBufferSize * Byte.BYTES / POSITION_SIZE));
-        estimatedOffsetBufferMaxCapacity = (int) (targetBufferSize * Integer.BYTES / POSITION_SIZE);
+        setEstimatedNullsBufferMaxCapacity((int) (targetBufferSize * Byte.BYTES / POSITION_SIZE * GRACE_FACTOR_FOR_MAX_BUFFER_CAPACITY));
```

Comment:
nit: perhaps, refactor to extract a helper method to avoid copy-paste

Reply:
@mbasmanova did you mean something like this?
```
setEstimatedNullsBufferMaxCapacity(getEstimatedBufferMaxCapacity(targetBufferSize, Byte.BYTES, POSITION_SIZE));
estimatedValueBufferMaxCapacity = getEstimatedBufferMaxCapacity(targetBufferSize, Byte.BYTES, POSITION_SIZE);
```
and in AbstractBlockEncodingBuffer:

```
protected static int getEstimatedBufferMaxCapacity(double targetBufferSize, int unitSize, int positionSize)
    {
        return (int) (targetBufferSize * unitSize / positionSize * GRACE_FACTOR_FOR_MAX_BUFFER_CAPACITY);
    }
```

Reply:
@yingsu00 Yes, this might reduce copy-paste and make it easier to read and ensure we don't forget GRACE_FACTOR_FOR_MAX_BUFFER_CAPACITY somewhere.

Reply:
@mbasmanova Hi Masha, I just updated the PR with a new commit `e8511df636 Refactor buffer max capacity calculation`. Thank you again, and happy long weekend!